static/js/components/CommentTree.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import React from "react";
+import R from "ramda";
+import moment from "moment";
+
+import Card from "../components/Card";
+
+import type { Comment } from "../flow/discussionTypes";
+
+const renderTopLevelComment = (comment: Comment, idx: number) =>
+  <Card key={idx}>
+    <div className="top-level-comment">
+      {renderComment(0, comment)}
+    </div>
+  </Card>;
+
+const commentClassName = depth => depth > 0
+  ? "comment indented"
+  : "comment";
+
+const renderComment = R.curry((depth: number, comment: Comment) =>
+  <div className={commentClassName(depth)} key={`${depth}${comment.created}`}>
+    <div className="author-info">
+      <a className="author-name">
+        {comment.author_id}
+      </a>
+      <div>
+        {moment(comment.created).fromNow()}
+      </div>
+    </div>
+    <div>
+      {comment.text}
+    </div>
+    { R.map(renderComment(depth + 1), comment.replies) }
```

Comment:
With the current approach, every comment element apart from the top-level comments has this 'indented' class added to it. If we wrap all of a comments replies into some parent element (instead of rendering them in succession as we're doing here) we can take care of the indent for all reply elements by adding the indent on that parent element. So we could do something like...

```javascript
<div className="replies">
  { R.map(renderComment(depth + 1), comment.replies) }
</div>
```

... and the indentation could be defined for `.replies`. We wouldn't need the 'indented' class at all, so we wouldn't need to calculate the class name or repeat it for a large number of elements